### PR TITLE
🔊 Collect computed and perf entry durations

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

In the context of the investigation of resources with  long durations, I want to assess the performance entry durations correctness.
To do it, I want to colllect xhr and fetch computed durations and corresponding performance entry durations to identify:

- happens when we got a the duration from performance entry, computed duration or both
- compare entry duration with computed duration that could highlight an issue from the browser

## Changes

Adds `resource._dd.computed_duration` and `resource._dd.entry_performance_duration`
Under feature flag resource_durations

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts [hunk 1]
```diff
@@ -84,7 +87,17 @@ function processRequest(
...
+    isExperimentalFeatureEnabled('resource_durations')
+      ? {
+          _dd: {
+            computed_duration: duration,
+            performance_entry_duration: correspondingTimingOverrides
+              ? correspondingTimingOverrides.resource.duration
+              : undefined,
+          },
+        }
+      : undefined
```
Comment (liywjl) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:100]: **💬 suggestion:** maybe lift this into a variable before
 `const resourceEvent = combine(` 
WDYT?

Comment (liywjl) [reply packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:100]: **💭 thought:** I added a diff and diff percentage variable on another [PR](https://github.com/DataDog/browser-sdk/pull/1810), is that something that you think might be useful?

Comment (amortemousque) [reply packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts]: I lifted it. 
I'd say that without your fix, I could have fallen into wrong conclusions, comparing computed and entry resource duration. So in that sense it helps me doing my investigation :)

**Review by liywjl — APPROVED** [review]

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts [hunk 2]
```diff
@@ -68,13 +69,25 @@ function processRequest(
...
+  const computedAndEntryDurations = isExperimentalFeatureEnabled('resource_durations')
+    ? {
+        _dd: {
+          computed_duration: duration,
+          performance_entry_duration: correspondingTimingOverrides
+            ? correspondingTimingOverrides.resource.duration
+            : undefined,
+        },
+      }
+    : undefined
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:82]: **💬 suggestion:** ‏what about extracting this logic in a function like the other computations?

File: packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts [hunk 3]
```diff
@@ -68,13 +69,25 @@ function processRequest(
...
   const responseDurationInfo = computeResponseDurationInfo(request)
 
+  const duration = toServerDuration(request.duration)
+  const computedAndEntryDurations = isExperimentalFeatureEnabled('resource_durations')
+    ? {
+        _dd: {
+          computed_duration: duration,
+          performance_entry_duration: correspondingTimingOverrides
+            ? correspondingTimingOverrides.resource.duration
+            : undefined,
```
Comment (bcaudan) [inline packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts:79]: **💬 suggestion:** ‏not sure how you plan to analyse those data but collecting directly the difference between the two may be easier to analyse.

**Review by BenoitZugmeyer — APPROVED** [review]

**Review by bcaudan — APPROVED** [review]

**Pull request merged**